actors/evm/src/interpreter/system.rs

```diff
@@ -262,13 +271,20 @@ impl<'r, RT: Runtime> System<'r, RT> {
                 "EIP-3541: Contract code starting with the 0xEF byte is disallowed.".into(),
             ));
         }
-        let k = self
+
+        let code_hash = multihash::Multihash::wrap(
+            SupportedHashes::Keccak256 as u64,
+            &self.rt.hash(SupportedHashes::Keccak256, bytecode),
+        )
+        .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to hash bytecode with keccak")?;
+
+        let cid = self
             .rt
             .store()
             .put(Code::Blake2b256, &Block::new(IPLD_RAW, bytecode))
             .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to write bytecode")?;
-        self.bytecode = Some(k);
-        Ok(k)
+        self.bytecode = Some((cid, code_hash));
```

Comment:
At this point, I'd consider introducing a very simple `Code` type that includes the bytecode + hash.